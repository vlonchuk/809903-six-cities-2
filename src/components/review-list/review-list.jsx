import React from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import ReviewItem from './../review-item/review-item.jsx';
import ReviewAdd from './../review-add/review-add.jsx';
import {MAX_COMMENTS_NUMBER} from '../../consts/constraints.js';

const ReviewList = ({hotelId, comments, isAuthorizationRequired}) => {
  const viewComments = comments.slice(0, MAX_COMMENTS_NUMBER);
  return <section className="property__reviews reviews">
    <h2 className="reviews__title">Reviews &middot;
      <span className="reviews__amount">{comments.length}</span>
    </h2>
    <ul className="reviews__list">
      {viewComments.map((el, i) => <ReviewItem review={el} key={i}/>)}
    </ul>
    {isAuthorizationRequired ? null : <ReviewAdd hotelId={hotelId}/>}
  </section>;
};

ReviewList.propTypes = {
  hotelId: PropTypes.number.isRequired,
  comments: PropTypes.array.isRequired,
  isAuthorizationRequired: PropTypes.bool.isRequired,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  isAuthorizationRequired: state.isAuthorizationRequired,
  comments: state.comments,
});

const ReviewListWrapped = connect(mapStateToProps)(ReviewList);
export {ReviewList};
export default ReviewListWrapped;
