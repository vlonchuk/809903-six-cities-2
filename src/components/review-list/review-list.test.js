import React from 'react';
import renderer from 'react-test-renderer';
import {ReviewList} from './review-list';
import reviewItem from './../../mocks/review-item.js';

jest.mock(`./../review-add/review-add.jsx`, () => jest.fn().mockReturnValue(null));

it(`ReviewList correctly renders after relaunch`, () => {
  const tree = renderer
    .create(
        <ReviewList
          hotelId={1}
          comments={[reviewItem]}
          isAuthorizationRequired={false}
          review={reviewItem}
        />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
