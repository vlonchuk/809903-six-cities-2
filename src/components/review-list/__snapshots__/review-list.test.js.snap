// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList correctly renders after relaunch 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â·
    <span
      className="reviews__amount"
    >
      1
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height={54}
            src="img/1.png"
            width={54}
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Oliver.conner
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-05-08"
        >
          May 2019
        </time>
      </div>
    </li>
  </ul>
</section>
`;
